registry/registry_test.go

```diff
@@ -0,0 +1,317 @@
+	})
+
+	cl := NewClient(&client.Harbor{Products: p, Transport: nil}, authInfo)
+	r, err := cl.GetRegistry(ctx, name)
+
+	p.AssertExpectations(t)
+	assert.Nil(t, r)
+	if assert.Error(t, err) {
+		require.IsType(t, &ErrRegistryInternalErrors{}, err)
+	}
+}
+
+func TestRESTClient_GetRegistry_ErrRegistryNotProvided(t *testing.T) {
+	p := &mocks.MockClientService{}
+	cl := NewClient(&client.Harbor{Products: p, Transport: nil}, authInfo)
+	r, err := cl.GetRegistry(context.Background(), "")
+
+	p.AssertExpectations(t)
+	assert.Nil(t, r)
+	if assert.Error(t, err) {
+		assert.IsType(t, &ErrRegistryNotProvided{}, err)
+	}
+}
+
+func TestRESTClient_DeleteRegistry(t *testing.T) {
+	ctx := context.Background()
+	registry := &model.Registry{
+		CreationTime: "",
+		Credential:   nil,
+		Description:  "",
+		ID:           10,
+		Insecure:     false,
+		Name:         "restregistry",
+		Status:       "",
+		Type:         "harbor",
+		UpdateTime:   "",
+		URL:          "http://foo.bar",
+	}
+
+	p := &mocks.MockClientService{}
+	p.On("GetRegistries",
+		&products.GetRegistriesParams{
+			Name:    &registry.Name,
+			Context: ctx,
+		}, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc"),
+	).Return(&products.GetRegistriesOK{
+		Payload: []*model.Registry{registry},
+	}, nil)
+	p.On("DeleteRegistriesID",
+		&products.DeleteRegistriesIDParams{
+			ID:      registry.ID,
+			Context: ctx,
+		}, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc"),
+	).Return(&products.DeleteRegistriesIDOK{}, nil)
+
+	cl := NewClient(&client.Harbor{Products: p, Transport: nil}, authInfo)
+	err := cl.DeleteRegistry(ctx, registry)
+
+	p.AssertExpectations(t)
+	assert.NoError(t, err)
+}
+
+func TestRESTClient_DeleteRegistry_NotFound(t *testing.T) {
+	ctx := context.Background()
+	registry := &model.Registry{
+		CreationTime: "",
+		Credential:   nil,
+		Description:  "",
+		ID:           10,
+		Insecure:     false,
+		Name:         "restregistry",
+		Status:       "",
+		Type:         "harbor",
+		UpdateTime:   "",
+		URL:          "http://foo.bar",
+	}
+
+	p := &mocks.MockClientService{}
+	p.On("GetRegistries",
+		&products.GetRegistriesParams{
+			Name:    &registry.Name,
+			Context: ctx,
+		}, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc"),
+	).Return(nil, &runtime.APIError{
+		OperationName: "",
+		Response:      "not found",
+		Code:          404,
+	})
+
+	cl := NewClient(&client.Harbor{Products: p, Transport: nil}, authInfo)
+	err := cl.DeleteRegistry(ctx, registry)
+
+	p.AssertExpectations(t)
+	if assert.Error(t, err) {
+		require.IsType(t, &runtime.APIError{}, err)
+		ty, _ := err.(*runtime.APIError)
+		assert.Equal(t, 404, ty.Code)
+		assert.Equal(t, "not found", ty.Response)
+	}
+}
+
+func TestRESTClient_DeleteRegistry_ErrRegistryNotProvided(t *testing.T) {
+	p := &mocks.MockClientService{}
+	cl := NewClient(&client.Harbor{Products: p, Transport: nil}, authInfo)
+	err := cl.DeleteRegistry(context.Background(), nil)
+
+	p.AssertExpectations(t)
```

Comment:
I think this can be left out, as there are no `mock.On` or `mock.Return` calls. As per the GoDoc of this method,
> AssertExpectations asserts that everything specified with On and Return was in fact called as expected.